server/src/utils/chess-game-util.js

```diff
@@ -0,0 +1,56 @@
+const calcExpectedScore = (_ratingA, _ratingB) => {
+  const ratingA = parseInt(_ratingA, 10);
+  const ratingB = parseInt(_ratingB, 10);
+  const score = 1 / (1 + (10 ** ((ratingB - ratingA) / 400)));
+  return parseFloat(score.toFixed(2));
+};
+
+const calcScores = (winner, pOneExpScore, pTwoExpScore) => {
+  let pOneScore;
+  let pTwoScore;
+
+  switch (winner) {
+    case 'player_one':
+      pOneScore = 1;
+      pTwoScore = 0;
+      break;
+    case 'player_two':
+      pOneScore = 0;
+      pTwoScore = 1;
+      break;
+    case 'draw':
+      pOneScore = 0.5;
+      pTwoScore = 0.5;
+      break;
+    default:
+      pOneScore = pOneExpScore;
+      pTwoScore = pTwoExpScore;
+  }
+
+  return { pOneScore, pTwoScore };
+};
+
+const calcEloRating = (oldRatingString, score, expectedScore) => {
+  const oldRating = parseInt(oldRatingString, 10);
+  let KFactor;
+  if (oldRating < 2100) {
+    KFactor = 32;
+  } else if (oldRating <= 2400) {
+    KFactor = 24;
+  } else {
+    KFactor = 16;
+  }
+  return Math.round(oldRating + KFactor * (score - expectedScore));
+};
+
+exports.calcRating = (playerOneRating, playerTwoRating, winner) => {
```

Comment:
es6

Reply:
does score change when it's a draw? Should it? If not we can simply return old results.

Reply:
In the rating system (Elo rating system) which we are using on draw,  rating is changed.